app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,211 @@
+        alertDialog.setNegativeButton(applicationContext.getString(R.string.no), null)
+        val alert = alertDialog.create()
+        alert.show()
+        val yes = alert.getButton(DialogInterface.BUTTON_POSITIVE)
+        val no = alert.getButton(DialogInterface.BUTTON_NEGATIVE)
+        yes.setTextColor(resources.getColor(R.color.md_blue_500))
+        no.setTextColor(resources.getColor(R.color.md_red_500))
+    }
+
+    override fun alertSuccess() {
+        val dialogClickListener = DialogInterface.OnClickListener { dialogInterface, i -> finish() }
+        val alertTitle = resources.getString(R.string.signup)
+        val alertMessage = resources.getString(R.string.signup_msg)
+        val successAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, null, resources.getString(R.string.ok), null, resources.getColor(R.color.md_blue_500))
+        successAlertboxHelper.showAlertBox()
+    }
+
+    override fun alertFailure() {
+        val dialogClickListener = DialogInterface.OnClickListener { dialogInterface, i ->
+            val intent = Intent(this@SignUpActivity, LoginActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+            startActivity(intent)
+            finish()
+        }
+        val dialogClickListenern = DialogInterface.OnClickListener { dialogInterface, i ->
+            val intent = Intent(this@SignUpActivity, ForgotPasswordActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+        val alertTitle = resources.getString(R.string.error_email)
+        val alertMessage = resources.getString(R.string.error_msg)
+        val failureAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, dialogClickListenern, resources.getString(R.string.ok), resources.getString(R.string.forgot_pass_activity), resources.getColor(R.color.md_blue_500))
+        failureAlertboxHelper.showAlertBox()
+    }
+
+    override fun alertError(message: String) {
+        val errorAlertboxHelper = AlertboxHelper(this@SignUpActivity, getString(R.string.unknown_host_exception), message, null, null, resources.getString(R.string.ok), null, Color.BLUE)
+        errorAlertboxHelper.showAlertBox()
+    }
+
+    override fun setErrorEmail() {
+        email?.error = getString(R.string.invalid_email)
+    }
+
+    override fun setErrorPass() {
+        password?.error = getString(R.string.error_password_matching)
+    }
+
+    override fun setErrorConpass(msg: String) {
+        confirm_password?.error = msg
+    }
+
+    override fun setErrorUrl() {
+        input_url?.error = getString(R.string.invalid_url)
+    }
+
+    override fun enableSignUp(bool: Boolean) {
+        sign_up?.isEnabled = bool
+    }
+
+    override fun isPersonalServer(): Boolean? {
+        return personal_server?.isChecked()
+    }
```

Comment:
Did not understand the usage of this method. You may remove this and pass personal_server?.isChecked() as parameter for signUp.